New Tile Class and whatDoYourSnakeEyesSee method

DHervochon opened the pull request:
Hasn't been tested but doesn't break anything, since nothing uses these yet.
Should be useful for cutting down time on pathfinding to food, and sets the foundation for panic mode.
Might want to move these into a classes file for Tile and parseBoard for snakeEyes.

DHervochon committed:
created Tile class

DHervochon committed:
created Tile class

DHervochon committed:
minichange 2 TIle, start of whatDoYourSnakeEyesSee

DHervochon committed:
minichange 2 TIle, start of whatDoYourSnakeEyesSee

DHervochon committed:
snakeEyes now starts from any desired tile

DHervochon committed:
snakeEyes now starts from any desired tile

DHervochon committed:
completed visible board method

DHervochon committed:
completed visible board method

DHervochon committed:
Merge remote-tracking branch 'origin/domBranch' into domBranch

DHervochon committed:
bugfix: need to visit tiles when adding to list

DimaMzk commented on app/main.py:
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
+    visited, instead of searching the entire list of visited tiles every time. Should be usable for other things.
+
+    Attributes:
+        x (integer): the x coordinate, should line up with our other methods
```
```suggestion
        x (int): the x coordinate, should line up with our other methods
```

DimaMzk commented on app/main.py:
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
+
+    Attributes:
+        x (integer): the x coordinate, should line up with our other methods
+        y (integer): the y coordinate etc
```
```suggestion
        y (int): the y coordinate etc
```

DimaMzk commented on app/main.py:
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
+    Attributes:
+        x (integer): the x coordinate, should line up with our other methods
+        y (integer): the y coordinate etc
+        visited (boolean): whether or not a tile has been visited, useful for some algorithms
```
```suggestion
        visited (bool): whether or not a tile has been visited, useful for some algorithms
```

DimaMzk commented on app/main.py:
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
+    This method finds the tiles it is possible to path to from any tiles coordinates, and the tiles that form walls around/in this area,
+    Then creates lists for this information.
+    Args:
+        pathBoard (array): integer representation of current board 
```
```suggestion
        pathBoard (list): integer representation of current board 
```

DimaMzk commented on app/main.py:
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
+    Then creates lists for this information.
+    Args:
+        pathBoard (array): integer representation of current board 
+        xPos(int): the x coordinate of the tile we want to search from
```
```suggestion
        xPos (int): the x coordinate of the tile we want to search from
```

DimaMzk commented on app/main.py:
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
+    Args:
+        pathBoard (array): integer representation of current board 
+        xPos(int): the x coordinate of the tile we want to search from
+        yPos(int): the y coordinate of the tile we want to search from
```
```suggestion
        yPos (int): the y coordinate of the tile we want to search from
```

AGuyNamedTate reviewed (changes requested):
Needs to pull changes from other branch. Currently breaks because of the printBoard function being moved to another file.

AGuyNamedTate commented on app/main.py:
```diff
@@ -305,57 +305,113 @@ def cardinal(converted_data, path):
+    """
+    board_width = len(pathBoard[0])
+    board_height = len(pathBoard)
+    allBoardTiles = [[None for x in range(board_width)] for y in range(board_height)] 
```
Please review variable name to reflect python naming conventions (all_board_tiles)

AGuyNamedTate reviewed (changes requested):
Still requires rebase

DHervochon committed:
moving things

new tile file for Tile class, moved whatDoYourSnakeEyesSee to parseBoard file, added imports for Tile class to main and parseBoard
